cubic-server/world-storage/Level.hpp

```diff
@@ -0,0 +1,47 @@
+#include <unordered_map>
+#include <functional>
+
+#include "ChunkColumn.hpp"
+
+#ifndef LEVEL_HPP
+#define LEVEL_HPP
+
+struct _2d_pos {
+    int x;
+    int z;
+
+    bool operator==(const _2d_pos& other) const {
+        return x == other.x && z == other.z;
+    }
+};
+
+template<>
+struct std::hash<_2d_pos>
+{
+    std::size_t operator()(_2d_pos const& pos) const noexcept
+    {
+        std::size_t h1 = std::hash<int>{}(pos.x);
+        std::size_t h2 = std::hash<int>{}(pos.z);
+        return h1 ^ (h2 << 1);
```

Comment:
This sounds like collisions but I am not sure.

Reply:
what do you mean by collisions ?